packages/ses/README.md

```diff
@@ -503,6 +502,153 @@ system generated other diagnostic information hidden in side tables. The tamed
 console uses these side tables to output more informative diagnostics.
 [Logging Errors](./src/error/README.md) explains the design.
 
+## Security claims and caveats
+
+Provided that the `ses` implementation and its trusted compute base are
+correct, we claim that a host program (Figure 1) can evaluate a guest program
+(`program`) in a compartment after `lockdown` and that the program:
+
+- will initially only have access to one mutable object, `globalThis`,
+- specifically cannot modify any primordial object that it shares with the host
+  program
+- cannot perform any I/O, apart from I/O performed within the trusted compute
+  base, directly or indirectly,
+- and specifically cannot observe the passage of time at any resolution.
+
+However, such a program can:
+
+- execute for an indefinite amount of time,
+- allocate arbitrary amounts of memory,
+- detect the platform endianness,
+- in some JavaScript engines, observe the contents of the stack.
+  This may include sensitive information about the layout of files on the host
+  disk.
+  In cases where the stack is data-dependent, a guest can infer the data.
+  SES occludes the stack on V8 and SpiderMonkey, but can not on
+  JavaScriptCore.
+
+```js
+// Claims, Figure 1
+lockdown();
+const compartment = new Compartment();
+compartment.evaluate(program);
+```
+
+The trusted compute base includes:
+
+- the host hardware,
+- the host operating system,
+- the process memory manager,
+- a minimal implementation of JavaScript conforming to ECMAScript 262 as of
+  2021, providing no unspecified host behavior like the introduction of syntax
+  that when evaluated reveals a mutable object.
+  SES accounts for one such host behavior provided by Node.js, namely the `domain`
+  property on promises, by preventing the use of `ses` in concert with the
+  `domain` module.
+- Also, any attached debugger, and
+- any JavaScript that has executed in the same realm before the host calls
+  `lockdown`, including JavaScript that executes after `ses` initializes.
+
+If the host program (Figure 2) arranges for the compartment's `globalThis` to
+be frozen, we additionally claim that the host can evaluate any two programs
+(`program1` and `program2`) such that neither program will:
+
+- initially share *any* mutable objects.
+- be able to observe the relative passage of time of the other program,
+- specifically because they do not have access to any shared mutable state as
+  they would had they been given a reference to `Math.random()`.
+
+```js
+// Claims, Figure 2
+lockdown();
+const compartment = new Compartment();
+harden(compartment.globaThis);
+compartment.evaluate(program1);
+compartment.evaluate(program2);
+```
+
+However no such programs (`program`, `program1`, nor `program2`) can be useful
+as such.
```

Comment:
Nice! But not quite true. The host has access to the second one (completion value!), and it could be a useful pure function.